voltdbclient/node_conn.go

```diff
@@ -62,7 +63,7 @@ func newNodeConn(ci string) *nodeConn {
 		closeCh:    make(chan chan bool),
 		drainCh:    make(chan chan bool),
 		responseCh: make(chan *bytes.Buffer, maxResponseBuffer),
-		requests:   make(map[int64]*networkRequest),
+		requests:   &sync.Map{},
```

Comment:
Is the race introduced because now we submit a ProcedureInvocation from a different go routine other than the loop routine?
If we do need a current Map, this is a 1.9 feature, we need also bump the required go version accordingly.

Reply:
Agreed. We came across the race condition yesterday. I can use mutex to lock the map which will work with older versions of Go. But I think using the new features is good for the library.